### PR TITLE
Allow #[derive(...)] methods to generate multiple items, and other cleanup

### PR DESCRIPTION
**Pull request by erickt**
This extracts some of the minor cleanup patches from #23905.

### CODE REVIEW

File: src/libsyntax/ext/deriving/cmp/eq.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 // option. This file may not be copied, modified, or distributed
```
Comment (huonw) [inline src/libsyntax/ext/deriving/cmp/eq.rs]: The changes to this file are essentially all spurious, right? (i.e. just from git not realising that `totaleq.rs` was renamed to overwrite `eq.rs`?)

Comment (erickt) [inline src/libsyntax/ext/deriving/cmp/eq.rs]: Yes, that's correct.

**Pull request merged**